src/ml/neural_net/image_augmentation.cpp

```diff
@@ -14,6 +14,42 @@
 namespace turi {
 namespace neural_net {
 
+shared_float_array convert_to_shared_float_array(
+    std::vector<image_annotation> annotations_per_image) {
+  size_t offset = 0;
+  std::vector<float> ann(annotations_per_image.size() * 6);
+  for (size_t j = 0; j < annotations_per_image.size(); j++) {
+    offset = j * 6;
+    image_annotation annotation = annotations_per_image[j];
+    ann[offset] = annotation.identifier;
+    ann[offset + 1] = annotation.bounding_box.x;
+    ann[offset + 2] = annotation.bounding_box.y;
+    ann[offset + 3] = annotation.bounding_box.height;
+    ann[offset + 4] = annotation.bounding_box.width;
+    ann[offset + 5] = annotation.confidence;
+  }
+  shared_float_array data_to_augment =
+      shared_float_array::wrap(std::move(ann), {annotations_per_image.size() * 6});
```

Comment:
Should the shape here be `{annotations_per_images.size(), 6}`? That is, rank 2 not 1?